core-tests/shared/src/test/scala/zio/ZManagedSpec.scala

```diff
@@ -670,8 +670,8 @@ object ZManagedSpec
                            for {
                              fiber        <- canceler.fork
                              _            <- latch.await
+                             interruption <- withLive(fiber.interrupt)(_.timeout(5.seconds)).either
```

Comment:
What is semantic change here?
Before `fiber.interrupt` was running with `Clock.Live` and now only `.timeout` is running with it?
How does it help with flakiness?

Reply:
No, you're totally fine. Changing `provide(Clock.Live)` to `withLive` doesn't have any semantic effect, it is just trying to clean up the code a little.

The semantically significant change is moving the `ref.set(false)` effect to after the interruption. I am using the `Ref` so that the cancelation action is non-terminating and we can check that it is uninterruptible. But since it never terminates the finalizer will still run at the end of the `Managed`, and we want that to terminate so that we don't hang. The problem is that created a race condition, because if the `Ref` is set to `false` before the forked canceler checks the `whenM` condition the canceler will become terminating and our test will fail, causing the flakiness. We fix that by moving the effect to set the Ref to after we have checked if we can interrupt the canceler.

Does that make sense?

Reply:
Yes, it does!
I think the `.either` doesn't bring any value, E is Nothing here and `.either` doesn't help with defects.